### PR TITLE
Update partition split count when task status changes

### PR DESCRIPTION
**Pull request by dain**
The count must be updated whenever the task status changes or
the cluster may deadlock since split slots marked as free.

### CODE REVIEW

File: presto-main/src/main/java/com/facebook/presto/server/remotetask/HttpRemoteTask.java [hunk 1]
```diff
@@ -240,6 +240,9 @@ public HttpRemoteTask(Session session,
                 if (state.isDone()) {
                     cleanUpTask();
                 }
+                else {
+                    partitionedSplitCountTracker.setPartitionedSplitCount(getPartitionedSplitCount());
```
Comment (cberner) [inline presto-main/src/main/java/com/facebook/presto/server/remotetask/HttpRemoteTask.java]: Why is this necessary in addition to the call in `updateTaskInfo`?

Comment (dain) [inline presto-main/src/main/java/com/facebook/presto/server/remotetask/HttpRemoteTask.java]: Not sure I understand the question, but I'll take a shot.  `updateTaskInfo` is called when the scheduler adds work on the task.  This is a callback when we get a status update from the task.  Most importantly, this lets us know that work has finished.

**Comment by electrum** [comment]
Looks good

File: presto-main/src/main/java/com/facebook/presto/server/remotetask/HttpRemoteTask.java (hunk 1 above)
Comment (cberner) [inline presto-main/src/main/java/com/facebook/presto/server/remotetask/HttpRemoteTask.java]: Oh I see. I was confused because I thought this state change listener only fired on task state changes, but it actually fires on all task status changes
